Commands/AnnouncementCmds.cs

```diff
@@ -265,41 +269,52 @@ public async Task AnnounceBuildSlashCommand(SlashCommandContext ctx,
             }
         }
 
-        [Command("editannouncetextcmd")]
-        [TextAlias("editannounce")]
+        [Command("editannounce")]
         [Description("Edit an announcement, preserving the ping highlight.")]
-        [AllowedProcessors(typeof(TextCommandProcessor))]
+        [AllowedProcessors(typeof(SlashCommandProcessor))]
         [RequireHomeserverPerm(ServerPermLevel.Moderator)]
         public async Task EditAnnounce(
-            TextCommandContext ctx,
-            [Description("The ID of the message to edit.")] ulong messageId,
-            [Description("The short name for the role to ping.")] string roleName,
-            [RemainingText, Description("The new message content, excluding the ping.")] string content
+            SlashCommandContext ctx,
+            [Parameter("message"), Description("The ID of the message to edit.")] string messageId,
+            [SlashChoiceProvider(typeof(AnnouncementRoleChoiceProvider))]
+            [Parameter("role1"), Description("The first role to ping.")] string role1Name,
+            [SlashChoiceProvider(typeof(AnnouncementRoleChoiceProvider))]
+            [Parameter("role2"), Description("The second role to ping. Optional.")] string role2Name = null
         )
         {
-            DiscordRole discordRole;
-
-            if (Program.cfgjson.AnnouncementRoles.ContainsKey(roleName))
+            // Validate msg ID
+            DiscordMessage msg;
+            try
             {
-                discordRole = await ctx.Guild.GetRoleAsync(Program.cfgjson.AnnouncementRoles[roleName]);
-                await discordRole.ModifyAsync(mentionable: true);
-                try
-                {
-                    await ctx.Message.DeleteAsync();
-                    var msg = await ctx.Channel.GetMessageAsync(messageId);
-                    await msg.ModifyAsync($"{discordRole.Mention} {content}");
-                }
-                catch
-                {
-                    // We still need to remember to make it unmentionable even if the msg fails.
-                }
-                await discordRole.ModifyAsync(mentionable: false);
+                msg = await ctx.Channel.GetMessageAsync(Convert.ToUInt64(messageId));
             }
-            else
+            catch
             {
-                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That role name isn't recognised!");
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That message ID wasn't recognised!", ephemeral: true);
                 return;
             }
+            
+            if (msg.Author.Id != ctx.Client.CurrentUser.Id)
+            {
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That message ID wasn't recognised!", ephemeral: true);
```

Comment:
This branch can be more specific

```suggestion
                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That message wasn't sent by me, so I can't edit it!", ephemeral: true);
```